src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,41 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a deadline task.
+ * @author Tee Kok Siang
+ */
+public class Deadline extends Task {
+    private final String by;
+
+    /**
+     * Constructs a Deadline object.
+     *
+     * @param description Task description.
+     * @param by Task deadline.
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    /**
+     * Returns formatted deadline task information.
+     * It will be used to write into the file.
+     * @return Formatted deadline task information.
+     */
+    @Override
+    public String toFileString() {
+        String done = super.isDone ? "1" : "0";
```

Comment:
Perhaps the `"1"` and `"0"` can be defined as variables in `Task.class`? Then the subclasses can utilise these variables, helping to abstract this flag further since it is used in both `Deadline.class` and `Event.class`.

Reply:
Right, I have extracted that part to the `Task` class. Thanks for the suggestion.